modyn/supervisor/internal/triggers/model_downloader/model_downloader.py

```diff
@@ -0,0 +1,106 @@
+import io
+import json
+import logging
+import pathlib
+from typing import Optional
+
+import grpc
+import torch
+from modyn.common.ftp import download_trained_model
+from modyn.metadata_database.metadata_database_connection import MetadataDatabaseConnection
+from modyn.metadata_database.models import TrainedModel
+
+# pylint: disable-next=no-name-in-module
+from modyn.model_storage.internal.grpc.generated.model_storage_pb2 import FetchModelRequest, FetchModelResponse
+from modyn.model_storage.internal.grpc.generated.model_storage_pb2_grpc import ModelStorageStub
+from modyn.models.coreset_methods_support import CoresetSupportingModule
+from modyn.utils import dynamic_module_import
+from modyn.utils.utils import grpc_connection_established
+
+logger = logging.getLogger(__name__)
+
+
+# TODO(366) Unify similar code in trainer_server and evaluator. Create common.utils.ModelDownloader
```

Comment:
366 might be the wrong issue number. I guess it would be cleaner to add this downloader as a common utility beforehand, but we can also do it afterwards. But we should definitly do it :)

Reply:
ModelDownloader Issue #433.